server/tests/services/test_hpa_vpa_recommendations.py

```
@@ -0,0 +1,214 @@
+    assert "error" in result
+    if provider:
+        assert provider in result["error"]
+
+
+def test_missing_pr_reference_is_rejected_before_any_network_call(monkeypatch):
+    """Assert the closer is never reached, not merely that an error came back --
+    otherwise this passes for the wrong reason (the fake user's GitHub auth
+    lookup failing) and would keep passing if the guard were removed."""
+    calls = []
+    monkeypatch.setitem(
+        _CLOSERS, "github", lambda *a, **kw: calls.append(a) or {"success": True}
```

Ruff (0.16.0)

[warning] Unused lambda argument: `kw`

(ARG005)